packages/mobile/src/identity/securityCode.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  ActionableAttestation,
+  AttestationServiceSecurityCodeRequest,
+  AttestationsWrapper,
+} from '@celo/contractkit/lib/wrappers/Attestations'
+import { PhoneNumberHashDetails } from '@celo/contractkit/src/identity/odis/phone-number-identifier'
+import { Address } from '@celo/utils/src/address'
+import BigNumber from 'bignumber.js'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'identity/securityCode'
+
+function hashAddressToSingleDigit(address: Address): number {
+  return new BigNumber(address.toLowerCase()).modulo(10).toNumber()
+}
+
+export function getSecurityCodePrefix(issuerAddress: Address) {
+  return `${hashAddressToSingleDigit(issuerAddress)}`
+}
+
+export function extractSecurityCodeWithPrefix(message: string) {
+  const matches = message.match('\\s(\\d{8})\\s')
+  if (matches && matches.length === 2) {
+    return matches[1]
+  }
+  return null
+}
+
+export async function getAttestationCodeForSecurityCode(
+  attestationsWrapper: AttestationsWrapper,
+  phoneHashDetails: PhoneNumberHashDetails,
+  account: string,
+  attestations: ActionableAttestation[],
+  securityCodeWithPrefix: string
+) {
+  const securityCodePrefix = parseInt(securityCodeWithPrefix[0], 10)
+  const lookupAttestations = attestations.filter(
+    (attestation) => hashAddressToSingleDigit(attestation.issuer) === securityCodePrefix
+  )
+
+  for (const attestation of lookupAttestations) {
+    // Try to recover the full attestation code from the matching issuer's attestation service
+    try {
+      const message = await requestValidator(
+        attestationsWrapper,
+        account,
+        phoneHashDetails,
+        attestation,
+        securityCodeWithPrefix.substr(1) // remove prefix
+      )
+      return message
+    } catch (e) {
+      Logger.warn(
+        TAG + '@getAttestationCodeForSecurityCode',
+        'Getting attestation code for security code error: ',
+        e
+      )
```

Comment:
wonder if we should differentiate between an expected error "wrong code" and others which we should log as Logger.error?